test/unified_malloc_framework/memoryPool.hpp

```diff
@@ -251,6 +258,25 @@ TEST_P(umfPoolTest, multiThreadedMallocFreeRandomSizes) {
     }
 }
 
+TEST_P(umfMemTest, outOfMem) {
+    static constexpr size_t allocSize = 16;
+    auto hPool = pool.get();
+
+    std::vector<void *> allocations;
+
+    while (true) {
+        allocations.emplace_back(umfPoolMalloc(hPool, allocSize));
+        if (umfPoolGetLastAllocationError(hPool) ==
```

Comment:
nit: you could add an assert to make sure that `umfPoolGetLastAllocationError(hPool)` is either STATUS_OK or ERROR_OUT_OF_HOST_MEMORY (we do no expect any other error).